### DISCUSSION TITLE
I want to ask, is the polymorphism of v5 finally stored in a table?

### DISCUSSION THREAD

**Discussion by Lydanne** [mikro-orm/mikro-orm#2503] [Q&A]
https://mikro-orm.io/docs/next/embeddables/#polymorphic-embeddables

My need is to realize the function of multiple tables, that is, multiple tables corresponding to one field, similar to mongo's refPath.

My realization：

```ts

@Entity()
export class Task extends CommonEntity {
  [EntityRepositoryType]: TaskRepo;

  @Property({ persist: false, hidden: true })
  private _attach = null;

  @Enum({ hidden: true })
  attachType?: AttachType; 

  @Property({ nullable: true, hidden: true })
  attachId: string; 

  get attach(): PromiseOrType<AttachForm> {
    if (!(this.attachType && this.attachId)) {
      return null;
    }
    if (this._attach) {
      return this._attach;
    }
    return this[EntityRepositoryType]?.getAttach<AttachForm>(
      this.attachType,
      this.attachId,
    ).then((data) => {
      this._attach = data;
      return data;
    });
  }

  @Property({ persist: false })
  set attach(attach: PromiseOrType<AttachForm>) {
    if (attach instanceof Promise) {
      attach.then((attach) => {
        this.attachType = attach.type;
        this.attachId = attach.id;
        this._attach = attach;
      });
    } else if (attach instanceof TaskAttach) {
      this.attachType = attach.type;
      this.attachId = attach.id;
      this._attach = attach;
    } else {
      throw new BusinessException(TaskErrcode.TYPE, 'not TaskAttach');
    }
  }
}

@Repository(Task)
export class TaskRepo extends CommonEntityRepository<Task> {
  getAttach<T>(attachType: AttachType, attachId: string): Promise<T> {
    return this.em.getRepository(attachType).findOne(attachId) as Promise<T>;
  }
}
```

**Comment by Lydanne** [comment]
I want to find the data I want to check through `attachId` and `attachType`.
`attachType` determine entity name
`attachId` determine entity data

**Comment by Lydanne** [comment]
I thought of a way that can be implemented in 5.0, but I still hope to support natively, if the following two features are used to implement the document, it will be nested one more layer

#1932
#2426

```ts
enum AnimalType {
  CAT,
  DOG,
}

@Entity()
class Cat {

  @Property({ nullable: true })
  canMeow?: boolean = true;

  constructor(name: string) {
    super();
    this.type = AnimalType.CAT;
    this.name = name;
  }

}

@Entity()
class Dog {
  @Property({ nullable: true })
  canBark?: boolean = true;

  constructor(name: string) {
    super();
    this.type = AnimalType.DOG;
    this.name = name;
  }

}


@Embeddable({ abstract: true, discriminatorColumn: 'type' })
abstract class AnimalProxy {

  @Enum(() => AnimalType)
  type!: AnimalType;

}

@Embeddable({ discriminatorValue: AnimalType.Cat })
class CatProxy extends AnimalProxy {
    @ManyToOne(() => Cat)
    pet!:Cat;
}


@Embeddable({ discriminatorValue: AnimalType.DOG })
class DogProxy extends AnimalProxy {
    @ManyToOne(() => Dog)
    pet!:Dog;
}


@Entity()
class Owner {

  @PrimaryKey()
  id!: number;

  @Property()
  name!: string;

  @Embedded(() => [CatProxy, DogProxy])
  proxy!: CatProxy | DogProxy;

}
```

I don't want to proxy this

**Comment by B4nan** [comment]
Doing `this[EntityRepositoryType]` is a huge misunderstanding of what that is. It is a type reference for strict type checks only, there is no value, never will be. 

And nope, polymorphic relations are not supported, the issue is here: #706

**Reply by Lydanne** [reply]
https://github.com/mikro-orm/mikro-orm/discussions/2503#discussioncomment-1777549
The problem I care more about is this. I don’t want to use embedded documents, but polymorphic embedded entities.

**Reply by B4nan** [reply]
As said, polymorphinc relations are not supported.

**Reply by Lydanne** [reply]
I want to use it in v5 ?

**Reply by B4nan** [reply]
Ok, so for the third time, polymorphic relations are not supported, the issue is #706, it is open, that means it is not implemented - not in v4, not in v5, nowhere.

**Reply by Lydanne** [reply]
Oh, okay, I thought ticking here means supporting

![image](https://user-images.githubusercontent.com/39021696/145385958-a44d5bf4-809f-4e98-b292-c5e94e8963fb.png)

**Reply by B4nan** [reply]
it means what it says - supporting entities inside embeddables, not supporting polymorphic relationships

**Reply by Lydanne** [reply]
I understand that ， What I want to know is, will polymorphic relationships be supported later

**Reply by Lydanne** [reply]
Rather than an embedded polymorphism
